core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -216,14 +216,14 @@ void HIPInternal::initialize(int hip_device_id, hipStream_t stream) {
     m_maxBlock = hipProp.maxGridSize[0];
 
     // theoretically, we can get 40 WF's / CU, but only can sustain 32
+    // see
+    // https://github.com/ROCm-Developer-Tools/HIP/blob/a0b5dfd625d99af7e288629747b40dd057183173/vdi/hip_platform.cpp#L742
     m_maxBlocksPerSM = 32;
     // FIXME_HIP - Nick to implement this upstream
-    m_regsPerSM          = 262144 / 32;
-    m_shmemPerSM         = hipProp.maxSharedMemoryPerMultiProcessor;
-    m_maxShmemPerBlock   = hipProp.sharedMemPerBlock;
-    m_maxThreadsPerSM    = m_maxBlocksPerSM * HIPTraits::WarpSize;
-    m_maxThreadsPerBlock = hipProp.maxThreadsPerBlock;
-
+    m_regsPerSM        = 262144 / m_maxBlocksPerSM;
```

Comment:
Can you add a comment that explains where 262144 comes from. Also is the `FIXME` still necessary?

Reply:
Good catch @Rombur, I was way off here. regsPerMultiprocessor isn't defined in HIP (yet, pushing the runtime people on it), but from the [ISA Document](https://www.olcf.ornl.gov/wp-content/uploads/2019/10/ORNL_Application_Readiness_Workshop-AMD_GPU_Basics.pdf), see the diagram in Sec 2.2.:

256 VGPR/SIMD * 4 SIMD/CU = 1024 VGPR/CU.

Reply:
Ok, figured it out -- they report in terms of registers (i.e., not 32 or 64-wide vector registers).
So here, this would be:

256 VGPR/SIMD * 4 SIMD/CU * 64 registers/VGPR = 65536 registers/CU